metricflow/time/time_granularity_solver.py

```diff
@@ -151,87 +72,65 @@ def validate_time_granularity(
+        valid_group_by_elements = self._semantic_manifest_lookup.metric_lookup.linkable_set_for_metrics(
+            metric_references=metric_references,
+        )
+        result: Dict[PartialTimeDimensionSpec, TimeDimensionSpec] = {}
         for partial_time_dimension_spec in partial_time_dimension_specs:
-            # Handle local time dimensions
-            if partial_time_dimension_spec.entity_links == ():
-                _, min_time_granularity_for_querying = self.local_dimension_granularity_range(
-                    metric_references=metric_references,
-                    local_time_dimension_reference=TimeDimensionReference(
-                        element_name=partial_time_dimension_spec.element_name
-                    ),
-                )
-
+            minimum_time_granularity: Optional[TimeGranularity] = None
+            for path_key in valid_group_by_elements.path_key_to_linkable_dimensions:
                 if (
-                    partial_time_dimension_spec.element_name == metric_time_dimension_reference.element_name
-                    and time_granularity
+                    path_key.element_name == partial_time_dimension_spec.element_name
+                    and path_key.entity_links == partial_time_dimension_spec.entity_links
+                    and path_key.time_granularity is not None
                 ):
-                    if time_granularity < min_time_granularity_for_querying:
-                        raise RequestTimeGranularityException(
-                            f"Can't use time granularity for time dimension '{metric_time_dimension_reference} since "
-                            f"the minimum granularity is {min_time_granularity_for_querying}"
-                        )
-                    replacement_dict[partial_time_dimension_spec] = TimeDimensionSpec(
-                        element_name=partial_time_dimension_spec.element_name,
-                        entity_links=(),
-                        time_granularity=time_granularity,
-                    )
-                else:
-                    replacement_dict[partial_time_dimension_spec] = TimeDimensionSpec(
-                        element_name=partial_time_dimension_spec.element_name,
-                        entity_links=(),
-                        time_granularity=min_time_granularity_for_querying,
+                    minimum_time_granularity = (
+                        path_key.time_granularity
+                        if minimum_time_granularity is None
+                        else min(minimum_time_granularity, path_key.time_granularity)
                     )
-            # Handle joined time dimensions.
-            else:
-                # TODO: For joined time dimensions, also compute the minimum granularity for querying.
-                replacement_dict[partial_time_dimension_spec] = TimeDimensionSpec(
+
+            if minimum_time_granularity is not None:
```

Comment:
I can't see it in the PR but I'm assuming this is indented inside of the outer for loop, so we build a dict for each partial_time_dimension_spec. Stupid proportional fonts....

Reply:
That is correct.